client/src/components/Board/Settings/MembersList.js

```diff
@@ -0,0 +1,204 @@
+import React, { useState, useEffect } from "react";
+import { SERVER_URL } from "./../../../config";
+import {
+  Typography,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Chip,
+  Switch,
+  Paper,
+  CircularProgress
+} from "@material-ui/core";
+import "./settings.css";
+import "./../../../App.css";
+import axios from "axios";
+import { useSnackBar } from "./../../../customHooks";
+import Icon from "@material-ui/core/Icon";
+import Modal from "./../../CommonComponents/Modal";
+import Button from "./../../Button";
+import constants from "./../../../constants";
+
+const MemberList = function(props) {
+  const [members, setMembers] = useState([]);
+  const [showLoader, setShowLoader] = useState(true);
+  const { openSnackBar } = useSnackBar();
+  const [ openConfirmationModal, setOpenConfirmationModal ] = useState(false);
+  const [ memberToBeDelete, setMemberToBeDelete ] = useState({});
+  
+  useEffect(() => {
+    fetchBoardMembers();
+  }, [props]);
+
+  async function fetchBoardMembers() {
+    try {
+      const result = await axios(SERVER_URL + "/board/members/86");
+      setMembers(result.data.data);
+      setShowLoader(false);
+    }
+    catch(error){
+      const { isSuccess, message } = error.response.data;
+      if (!isSuccess) {
+        openSnackBar('error', message);
+      }
+    }
+  }
+
+  const updateBoardMember = async data => {
+    try {
+      const response = await axios({
+        url: `${SERVER_URL}/board/member/86`,
+        method: "patch",
+        data
+      });
+      const { isSuccess, message } = response.data;
+      openSnackBar('success', message);
+      fetchBoardMembers();
+    }
+    catch(error){
+      const { isSuccess, message } = error.response.data;
+      if (!isSuccess) {
+        openSnackBar('error', message);
+      }
+    }
+  };
+
+  const setMemberRole = (isChecked, index) => {
+    const boardMembers = Object.assign([], members);
+    if (isChecked) {
+      boardMembers[index].role = constants.ROLES.ADMIN;
+    } else {
+      boardMembers[index].role = constants.ROLES.USER;
+    }
```

Comment:
```js
    // somewhere in top
    const { SUPER_ADMIN, ADMIN, USER } = ROLES;

    ...
    ...

    // to setRole...
    boardMembers[index].role = isChecked ? ADMIN : USER;
```

Reply:
Please give comments if there anything wrong in logic. Every person coding style is different

Reply:
I think if the code is readable and optimize then it is good code

Reply:
Done